google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1beta2/st/STBigQueryStorageLongRunningWriteTest.java

```diff
@@ -0,0 +1,211 @@
+      Descriptors.DescriptorValidationException {
+    // Set up a default stream. Write to it for a long time, (a few minutes for now) and make
+    // sure that everything goes well.
+    String tableName = "JsonTableDefaultStream";
+    TableInfo tableInfo =
+        TableInfo.newBuilder(
+            TableId.of(DATASET, tableName),
+            StandardTableDefinition.of(
+                Schema.of(
+                    com.google.cloud.bigquery.Field.newBuilder(
+                        "test_str", StandardSQLTypeName.STRING)
+                        .build(),
+                    com.google.cloud.bigquery.Field.newBuilder(
+                        "test_numerics", StandardSQLTypeName.NUMERIC)
+                        .setMode(Field.Mode.REPEATED)
+                        .build(),
+                    com.google.cloud.bigquery.Field.newBuilder(
+                        "test_datetime", StandardSQLTypeName.DATETIME)
+                        .build())))
+            .build();
+    bigquery.create(tableInfo);
+    TableName parent = TableName.of(ServiceOptions.getDefaultProjectId(), DATASET, tableName);
+    try (JsonStreamWriter jsonStreamWriter =
+        JsonStreamWriter.newBuilder(parent.toString(), tableInfo.getDefinition().getSchema())
+            .createDefaultStream()
+            .setBatchingSettings(
+                StreamWriter.Builder.DEFAULT_BATCHING_SETTINGS
+                    .toBuilder()
+                    .setRequestByteThreshold(1024 * 1024L) // 1 Mb
+                    .setElementCountThreshold(2L)
+                    .setDelayThreshold(Duration.ofSeconds(2))
+                    .build())
+            .build()) {
+      for (int i = 0; i < 5; i++){
+        LOG.info("Sending a message");
+        // Ramping up the size increases the latency
+        JSONObject row = MakeJsonObject(1);
+        JSONArray jsonArr = new JSONArray(new JSONObject[] {row});
+
+        LocalDateTime start = LocalDateTime.now();
+        Date startTime = new Date();
+        //TODO(jstocklass): Make asynchronized calls instead of synchronized calls
+        ApiFuture<AppendRowsResponse> response = jsonStreamWriter.append(jsonArr, -1);
+        assertEquals(0, response.get().getAppendResult().getOffset().getValue());
+        Assert.assertFalse(response.get().getAppendResult().hasOffset());
+        Date finishTime = new Date();
+        LOG.info("Latency: ".concat(String.valueOf(finishTime.getTime() - startTime.getTime())).concat(" ms"));
+         // seems like 2 or 3 seconds on average
+        //LOG.info(String.valueOf((Math.abs(finish.getSecond())-start.getSecond())));
+      }
+
+      TableResult result =
+          bigquery.listTableData(
+              tableInfo.getTableId(), BigQuery.TableDataListOption.startIndex(0L));
+      Iterator<FieldValueList> iter = result.getValues().iterator();
+      FieldValueList currentRow;
+      for (int i = 0; i < 5; i++) {
+        currentRow = iter.next();
```

Comment:
You may want to assert that iter hasNext; the assertion failure might have a nicer error message than an exception if something goes wrong.

Reply:
I've added that in, that makes sense to me